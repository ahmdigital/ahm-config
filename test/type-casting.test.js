const config = require('../lib');

describe('ahm-config: type-casting', () => {
  const path = `${__dirname}/type-casting`;

  beforeAll(() => {
    process.env.a = '2';
    process.env.b = 'false';
    process.env.c = '3.45';
    process.env.d__e__f = '6.7';
  });

  afterAll(() => {
    delete process.env.a;
    delete process.env.b;
    delete process.env.c;
  });

  it('should return schema errors when normalise disabled', () =>
    new Promise((done) => {
      const onError = (err) => {
        expect(err.message).toEqual(
          "Config error: undefined is not of a type(s) integer. Value 'undefined' should be undefined.",
        );
        expect(err.meta).toMatchSnapshot()
        done();
      };
      config.make({ path, normalise: false, onError });
    }));

  it('should type cast string literals to appropriate types', () => {
    const onError = () => {
      throw new Error('It should not happen');
    };
    const store = config.make({ path, normalise: true, onError });
    expect(store.get('a')).toBe(2);
    expect(store.get('b')).toBe(false);
    expect(store.get('c')).toBe(3.45);
    expect(store.get('d:e:f')).toBe(6.7);
  });
});
