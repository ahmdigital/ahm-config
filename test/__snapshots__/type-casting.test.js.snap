// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ahm-config: type-casting should return schema errors when normalise disabled 1`] = `
Array [
  ValidationError {
    "argument": Array [
      "integer",
    ],
    "instance": "2",
    "message": "is not of a type(s) integer",
    "name": "type",
    "path": Array [
      "a",
    ],
    "property": "instance.a",
    "schema": Object {
      "required": true,
      "type": "integer",
    },
    "stack": "instance.a is not of a type(s) integer",
  },
  ValidationError {
    "argument": Array [
      "boolean",
    ],
    "instance": "false",
    "message": "is not of a type(s) boolean",
    "name": "type",
    "path": Array [
      "b",
    ],
    "property": "instance.b",
    "schema": Object {
      "required": true,
      "type": "boolean",
    },
    "stack": "instance.b is not of a type(s) boolean",
  },
  ValidationError {
    "argument": Array [
      "number",
    ],
    "instance": "3.45",
    "message": "is not of a type(s) number",
    "name": "type",
    "path": Array [
      "c",
    ],
    "property": "instance.c",
    "schema": Object {
      "required": true,
      "type": "number",
    },
    "stack": "instance.c is not of a type(s) number",
  },
  ValidationError {
    "argument": Array [
      "number",
    ],
    "instance": "6.7",
    "message": "is not of a type(s) number",
    "name": "type",
    "path": Array [
      "d",
      "e",
      "f",
    ],
    "property": "instance.d.e.f",
    "schema": Object {
      "required": true,
      "type": "number",
    },
    "stack": "instance.d.e.f is not of a type(s) number",
  },
]
`;
